Pipfile

```diff
@@ -5,12 +5,12 @@ name = "pypi"
 
 [packages]
 "rfc5424-logging-handler" = "*"
-argo-workflows = ">=2.0,<3.0"
+argo-workflows = ">=3.5"
 daiquiri = "*"
 requests = "*"
 sentry-sdk = "*"
-kubernetes = "==10.0.0"  # required pyyaml>=3.12
-openshift = "~=0.10.0"
+kubernetes = "~=11.0.0"  # required pyyaml>=3.12
```

Comment:
Will the update happen for user api?

Reply:
are you referring to the user-api connexion issue we faced earlier ?
connexion should work as pyyaml is at 5.3.1 version.

Reply:
maybe this can asnwer, from @harshad16:

as frido stated the issue was a mismatch of version for pyyaml with respect to kubernetes/openshift/argo.
we required version of pyyaml version around >=3.12,  as 4.1.0 had cve issue.we tried to skip it. 
https://github.com/advisories/GHSA-rprw-h62v-c2w7
https://github.com/thoth-station/common/commit/eb5a4809af2f27568a61e2cd4b544b4f3191541f


however at that time kubernetes API v11.0.0 wasn't support by argo-python-client 
as i see now the latest update of v3.1.4 argo-python-client only has support for v11.0.0 available
more details: https://github.com/thoth-station/kebechet/issues/335#issuecomment-607468027


now we can use it by updating common requirement https://github.com/thoth-station/common/blob/master/requirements.txt#L12
so that it can use the latest version of argo-python-client v3.1.4 and then we can update the rest as well.

Reply:
@fridex

Reply:
Tried to install thoth-common from this branch to user-api - just to verify if the resolution process works on user-api side and that the update of thoth-common will be possible. Getting this error:

```
ERROR: Could not find a version that matches argo-workflows<3.0,>=2.0,>=3.5 (from -r /tmp/pipenv287v420xrequirements/pipenv-s4jznlx_-constraints.txt (line 20))
Tried: 1.0.0, 1.0.0, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.3.0, 1.3.0, 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.1.1, 2.1.1, 2.1.2, 2.1.2, 2.1.3, 2.1.3, 2.1.4, 2.1.4, 3.0.0, 3.0.0, 3.0.2, 3.0.2, 3.0.3, 3.0.3, 3.1.0, 3.1.0, 3.1.1, 3.1.1, 3.2.0, 3.2.0, 3.5.0, 3.5.0, 3.5.1, 3.5.1
```

Reply:
Sorry, comment stayed in a pending state somehow.

Reply:
To reproduce:

```console
cd thoth-station/user-api
git pull
pipenv install 'git+https://github.com/pacospace/common@bump-versions#egg=thoth-common'
```

Reply:
that is weird as `argo-workflows` is requested only from thoth-common, I don't understand how the conflict arise.

Reply:
I changed both `Pipfile`, `Pipfile.lock` and `requirements.txt`

Reply:
Got the same error and I see the issue, that constraint comes from thoth-python which depends on thoth-common, therefore we need to updated thoth-common, then thoth-python, then we can update user-api. See: https://github.com/thoth-station/python/blob/9c4bdd1da86d7752cabc034ccb5cd852cc229a9e/Pipfile#L10

Reply:
The log says:

```
There are incompatible versions in the resolved dependencies:
  argo-workflows>=3.5 (from -r /tmp/pipenvl7kzev1lrequirements/pipenv-s95dfee1-constraints.txt (line 6))
  argo-workflows<3.0,>=2.0 (from thoth-common==0.19.0->thoth-python==0.10.1->-r /tmp/pipenvl7kzev1lrequirements/pipenv-s95dfee1-constraints.txt (line 18))
```

It sounds like Pipenv is not resolving correctly for Git releases. thoth-python does not pin thoth-common:

https://github.com/thoth-station/python/blob/9c4bdd1da86d7752cabc034ccb5cd852cc229a9e/requirements.txt#L3

Checking the dependency graph, all should be fine.

Reply:
Anyway, I would give this a try. We can merge... ACK @pacospace?

Reply:
let's do it!! Thanks for checking!!